### PR TITLE
Update actions/checkout in GitHub Actions workflows to v3

### PR DESCRIPTION
**Pull request by striezel**
Updates the `actions/checkout` action used in the GitHub Actions workflow to its newest major version.

Changes in [actions/checkout](https://github.com/actions/checkout):

> ## v3.1.0
> - Use @actions/core `saveState` and `getState`
> - Add `github-server-url` input
>
> ## v3.0.2
> - Add input `set-safe-directory`
>
> ## v3.0.1
> - Fixed an issue where checkout failed to run in container jobs due to the new git setting `safe.directory`
> - Bumped various npm package versions
>
> ## v3.0.0
>
> - Update to node 16

Still using v2 of `actions/checkout` will generate some warning like in this run: https://github.com/unicode-org/icu4x/actions/runs/3761518854

> Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout@v2, actions/cache@v2, actions-rs/cargo@v1.0.1

The PR will get rid of those warnings for `actions/checkout`, because v3 uses Node.js 16.

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

**Comment by striezel** [comment]
@Manishearth: Since those changes have been approved, is there anything that still blocks this from merging?

The `actions/checkout` action is used in most workflows of this repository, so it is responsible for the majority of the deprecation warnings. Or in other words: Merging this should get rid of most deprecation warnings related to GitHub actions in this repository.

**Comment by Manishearth** [comment]
Yeah don't worry, if it's approved that means it'll get merged eventually: I think I approved this before CI was fully green so I didn't merge it then.

**Comment by striezel** [comment]
Good to know. :+1:

**Pull request merged**
